java/fury-core/src/main/java/org/apache/fury/util/StringEncodingUtils.java

```diff
@@ -0,0 +1,362 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.fury.util;
+
+import static org.apache.fury.util.StringUtils.MULTI_CHARS_NON_LATIN_MASK;
+
+import org.apache.fury.memory.Platform;
+
+/** String Encoding Utils. */
+public class StringEncodingUtils {
+
+  public static int convertUTF16ToUTF8(char[] src, byte[] dst, int dp) {
```

Comment:
Maybe we need to add a javadoc to document that this method will be faster only when there are many assci chars in the string. Acutally I'm curious whether this method will be faster than JDK native implementation for other string

Reply:
The overall algorithm idea implemented by this method is based on the scalar version of simdutf, and the non-ascillary part is based on the implementation of fastjson2（The speedup ratio is twice of JDK）. Even in the all-Chinese scenario, it will be faster than the JDK implementation.